site/packages/aa-signers/fireblocks/constructor.md

```diff
@@ -0,0 +1,96 @@
+---
+outline: deep
+head:
+  - - meta
+    - property: og:title
+      content: FireblocksSigner • constructor
+  - - meta
+    - name: description
+      content: Overview of the constructor method on FireblocksSigner in aa-signers
+  - - meta
+    - property: og:description
+      content: Overview of the constructor method on FireblocksSigner in aa-signers
+---
+
+# constructor
+
+To initialize a `FireblocksSigner`, you must provide a set of parameters detailed below.
+
+## Usage
+
+::: code-group
+
+```ts [example.ts]
+import { FireblocksSigner } from "@alchemy/aa-signers";
+import { ChainId } from "@fireblocks/fireblocks-web3-provider";
+
+// instantiates using every possible parameter, as a reference
+const fireblocksSigner = new FireblocksSigner({
+  apiKey: "API_KEY",
+  privateKey: "PATH_TO_FILE_WITH_PRIVATE_KEY",
+  chainId: ChainId.SEPOLIA,
+  rpcUrl: "ALCHEMY_RPC_URL",
+  vaultAccountIds: ["VAULT_ID"],
+  apiBaseUrl: "FIREBLOCKS_API_URL",
+  note: "note",
+  pollingInterval: 1000,
+  oneTimeAddressesEnabled: true,
+  externalTxId: "1",
+  userAgent: "test-header",
+  assetId: "1",
+  logTransactionStatusChanges: true,
+  logRequestsAndResponses: true,
+  enhancedErrorHandling: true,
+  gaslessGasTankVaultId: 1,
+});
+```
+
+:::
+
+## Returns
+
+### `FireblocksSigner`
+
+A new instance of a `FireblocksSigner`.
+
+## Parameters
+
+### `params: FireblocksProviderConfig | { inner: FireblocksWeb3Provider }`
+
+You can either pass in a constructed `FireblocksWeb3Provider` object, or directly pass into the `FireblocksSigner` the `FireblocksProviderConfig` used to construct a `FireblocksWeb3Provider` object. These parameters are listed on the [Fireblocks repo](https://github.com/fireblocks/fireblocks-web3-provider/blob/main/src/types.ts#L48) as well.
+
+`FireblocksProviderConfig` takes in the following parameters:
```

Comment:
oh oof... this is gonna be a pain to have to do and maintain for each signer provider in the future. I can see the devex benefit of having this here, but once it goes out of sync it causes more harm than good. can we link out to their types or docs instead?

Reply:
yeah where possible. some SDKs don't expose this, so I have to copy-paste onto the types.ts file for the Signer (so we'll still be out of sink).

but yeah, will just link to code where possible. (saves me time too 🙏 )